src/OpenLoco/Ui/TextInput.cpp

```diff
@@ -13,7 +13,7 @@ namespace OpenLoco::Ui::TextInput
     // Common code from 0x0044685C, 0x004CE910
     bool InputSession::handleInput(uint32_t charCode, uint32_t keyCode)
     {
-        if ((charCode >= SDLK_SPACE && charCode < SDLK_F12) || (charCode >= 159 && charCode <= 255))
+        if ((charCode >= SDLK_SPACE && charCode < SDLK_DELETE) || (charCode >= 159 && charCode <= 255))
```

Comment:
It's coincidence that the new correct value here is 127, aka DEL in ascii, as 127 marks the end of the regular valid ascii range of values. Previously this && caught literally every key since `SDLK_F12` is a huge value, and hence always tried to append things to the buffer. Now the range is correct.